build-info-extractor-maven3/src/main/java/org/jfrog/build/extractor/maven/DependencyResolutionSpy.java

```diff
@@ -0,0 +1,104 @@
+package org.jfrog.build.extractor.maven;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.apache.maven.eventspy.AbstractEventSpy;
+import org.apache.maven.eventspy.EventSpy;
+import org.apache.maven.project.DependencyResolutionResult;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.graph.DependencyNode;
+import org.jfrog.build.extractor.BuildInfoExtractorUtils;
+
+import java.util.*;
+
+/**
+ * Listen to DependencyResolutionResult events and populate parents map for each dependency of the current module.
+ * Each parent is an array of strings contains the path from parent GAV to module GAV.
+ *
+ * @author yahavi
+ **/
+@Component(role = EventSpy.class)
+public class DependencyResolutionSpy extends AbstractEventSpy {
+
+    @SuppressWarnings("unused")
+    @Requirement
+    private BuildInfoRecorder buildInfoRecorder;
+
+    @Override
+    public void onEvent(Object event) {
+        if (event instanceof DependencyResolutionResult) {
+            DependencyResolutionResult result = (DependencyResolutionResult) event;
+            buildInfoRecorder.setDependencyParentsMaps(createDependencyParentsMap(result.getDependencyGraph()));
+        }
+    }
+
+    /**
+     * Create a map of dependency to parents.
```

Comment:
Let's ellaborate more to make it as easy as possble for the random reader to understand how the returned map is structured.
Let's also add an example, and use ti to exp,ain the structure.